character_database.gd

```diff
@@ -1,56 +1,206 @@
-		], 
-		PortraitView.instantiate() # will be random
-	)
+var negative_properties_secret: Array[PropertyData] = [
+	PropertyData.new(-1, "Had intimate relations with the Sphinx", true),
+	PropertyData.new(-1, "Called their girlfriend's period the 'Red Sea'", true),
+	PropertyData.new(-3, "Was actually three kids in a trenchcoat", true),
+	PropertyData.new(-4, "Was quite an infamous graffiti artist", true),
+	PropertyData.new(-6, "Was part of an organized crime family", true),
+	PropertyData.new(-7, "Was a vampire", true),
 ]
+var positive_properties_secret: Array[PropertyData] = [
+	PropertyData.new(1, "Their Secret Osiris gifts were always on point", true),
+	PropertyData.new(2, "Never bragged about his great-uncle who invented the wheel", true),
+	PropertyData.new(4, "Had a tattoo on their waist saying 'Anubis rulez'", true),
+	PropertyData.new(6, "Wrote the original script of the Lion King", true),
+	PropertyData.new(9, "They were Batman", true)
+]
+var firstnames = ["Ashur", "Tarmek", "Shumak", "Abasi", "Abubakar", "Akiiki", "Amenhotep", "Ba", "Baahir", "Baal",
+				  "Bab", "Babita", "Badi", "Bebti", "Bomani", "Bastet", "Chatuluka", "Chigaru", "Chione", "Dalila",
+				  "Dakarai", "Dendera", "Donkor", "Ebo", "Echidna", "Edjo", "Fenuku", "Femi", "Eshe", "Hanbal",
+				  "Hapi", "Hapu", "Hathor", "Hatshepsut", "Hu", "Issa", "Jafari", "Jamila", "Kamilah", "Kakra",
+				  "Kafele", "Kamuzu", "Kanika", "Kek", "Kissa", "Lapis", "Lateef", "Layla", "Lotus", "Luzige",
+				  "Lisimba", "Maat", "Madu", "Maibe", "Mafuane", "Menkaura", "Moshe", "Msamaki", "Mukamutara",
+				  "Naeem", "Nassor", "Nathifa", "Neema", "Niu", "Niut", "Nkuku", "Nun", "Nuru", "Oba", "Olabisi",
+				  "Omorose", "Onuris", "Osahar", "Paki", "Ptolemy", "Pili", "Radames", "Rashida", "Runihura", "Sabah",
+				  "Sabola", "Sagira", "Sanura", "Sebak", "Sefu", "Selma", "Sekhet", "Shukura", "Siti", "Sutekh",
+				  "Tabia", "Tahirah", "Tau", "Tarik", "Thabit", "Thutmose", "Uadjit", "Ufa", "Ur", "Urbi", "Usi",
+				  "Ur Atum", "Wamukota", "Walidah", "Yafeu", "Zahra", "Zahur", "Zalika", "Zaid"
+				]
+
+var lastnames = ["the Unworthy", "the Worst", "the Meh", "the Neutral", "the Underachiever", "the Robust",
+					  "the Coward", "the Keg", "the Vulture", "the Radiant", "the Ambitious", "the Valiant",
+					  "the Bright", "the Dull", "the Bland", "the Forgotten", "the Hardcore", "the Rigid",
+					  "the Fierce", "the Tenacious", "the Amiable", "the Fearless", "the Real", "the Vivacious",
+					  "the Flexible", "the Shrewd", "the Cunning", "the Brave", "the Two-Faced", "the Relentless",
+					  "the Reckless", "the Lone Wolf", "the Great", "the Awesome", "the Tall", "the Fearsome",
+					  "the Shy", "the Handsome", "the Short", "the Unlikely", "the Wholesome", "the Destined One",
+					  "the Unfortunate", "the Lucky", "the Amateur", "the Inappropriate", "the Cringe", "the Awkward",
+					  "the Morbid"
+					]
+
+var characters: Array[CharacterData] = get_characters(10)
```

Comment:
Nem teljesen értem a célját annak hogy összerakunk egy 10 karakterből álló tömböt.

A fejemben tök elégnek tűnik, hogy van egy get_next_character() fv. ami generál egy random karaktert.
```
return CharacterData.new(
           get_random_fullname(),
           get_random_properties(positive_properties_normal,3),
           get_random_properties(negative_properties_normal,3), 
           null
)
```
Magát, hogy ez egy tizes tömb, amúgy sem használjuk sehol, mert a tinder screen csak kiválaszt random belőle egyet.
És így ki lehet hagyni a `characters` lokális tömböt, a `get_characters()` függvényt a _characted_data.gd_-ből és a `current_level_10_characters` lokális tömböt a tinder screenről.

Reply:
ezzel egyetértek, nem tudtam h nálatok feature branchen kell-e az tömbként érkezik ezért pluszban odakészítettem :D

Reply:
félreértés volt az egész én azért csináltam ezt a character_db-t eleve mert azt hittem a karakterek nem *teljesen* randomok lesznek, hanem majd a többiek kitalálják őket kb. de ha azon az úton megyünk végül hogy igen, és szerintem így egyszerűbb is amúgy, akkor az egész array felesleges igen